### DISCUSSION TITLE
Label creation with label printers

### DISCUSSION THREAD

**Discussion by Teketa** [Part-DB/Part-DB-server#746] [Ideas]
First of, i would like to thank everyone involved with this amazing project!

My wish is that there could be some form of integration with label printers, like the Brother P-Touch Series, so that labels could be printed directly from the site. Maybe though webUSB API and this tool https://github.com/Ircama/PT-P300BT. 

If that is too wild, then just being able to export the labels in PNG and then using the linked tool, would be way more convenient.
 
Would love to hear what everyone thinks about this!

**Comment by SapuSeven** [comment]
Note that the WebUSB API is quite controversial and not all browsers support it - see https://developer.mozilla.org/en-US/docs/Web/API/USB

**Comment by salfter** [comment]
Older Zebra label printers are pretty cheap on eBay, and are easy to get running under Linux or Windows.  I have an LP2844 and a QL420 Plus.  The LP2844 is a desktop printer that connects over USB, while the QL420 Plus is a portable printer that can run on battery and connects over USB or WiFi.  A quick check of eBay found some available for under $50.

**Comment by alextrical** [comment]
Also in the same boat wanting to use a label printer.

I have a Brother QL-500 that is supported through Linux as a CUPS device or can be printed to through this project https://github.com/trialinfo/ptouch-driver
Or via command line or API with https://github.com/HenrikBengtsson/brother-ptouch-label-printer-on-linux

I also have a Dymo LabelManager PnP and a Dymo LabelManager PCII that can be printed to on Linux via CUPS or command line through https://github.com/labelle-org/labelle

I will take a look into what is needed to set these up next week as I'm looking to sort out my stock of components in the near future, and having this support in part DB would be ideal

**Reply by alextrical** [reply]
I think the best I will be able to do is setup a Raspberry Pi , as a CUPS server. That allows me to setup the Brother QL-500 to print 29x90mm paper labels.

I've also managed to get a cheap reciept printer setup on the CUPS server in case I can think of a use for it

I'm waiting for the Dymo LabelManager's to arrive for setup and testing.
